### PR TITLE
Redoing how we deal with comments to fix some edge cases

### PR DESCRIPTION
**Pull request by belav**

### CODE REVIEW

File: Src/CSharpier.Tests/FormattingTests/TestFiles/Comments_EdgeCase.expected.cst [hunk 1]
```diff
@@ -6,21 +6,42 @@ public class ClassName
 
     public void Method()
     {
-        /* multiline
-         *  multiline
-         */
+        /* this doesn't work well because the indentation is changing
```
Comment (shocklateboy92) [inline Src/CSharpier.Tests/FormattingTests/TestFiles/Comments_EdgeCase.expected.cst:9]: Ugh... I'll be real this looks cursed

File: Src/CSharpier.Tests/FormattingTests/TestFiles/Comments_EdgeCase.expected.cst [hunk 2]
```diff
@@ -6,21 +6,42 @@ public class ClassName
...
 }
 
 public interface Interface
 {
     /// <summary>
     ///	</summary> tab here before <
     void Method();
+
+    /* this also doesn't work well because of turning 2 tabs into 4 spaces
+		and it keeps the tabs here, which is maybe what we want
```
Comment (shocklateboy92) [inline Src/CSharpier.Tests/FormattingTests/TestFiles/Comments_EdgeCase.expected.cst:37]: I think most editors will automatically turn tabs into spaces. So we'll probably receive spaces :shrug:

Comment (belav) [reply Src/CSharpier.Tests/FormattingTests/TestFiles/Comments_EdgeCase.expected.cst:37]: yeah this feels super edge case, there was one file in moq4 that had this issue.
